src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetLockService.cs

```diff
@@ -18,17 +18,24 @@ public sealed class NuGetLockService : INuGetLockService, IDisposable
     {
         private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
 
-        private readonly AsyncLocal<int> _lockCount = new AsyncLocal<int>();
+        public readonly AsyncLocal<int> LockCount = new AsyncLocal<int>();
```

Comment:
This makes `LocalCount.Value` externally mutable.  It would be better to keep the field private and add a readonly property that returns `_lockCount.Value`.